sensor_msgs/include/sensor_msgs/impl/point_cloud2_iterator.hpp

```diff
@@ -272,8 +273,9 @@ PointCloud2IteratorBase<T, TT, U, C, V>::PointCloud2IteratorBase(
   int offset = set_field(cloud_msg, field_name);
 
   data_char_ = &(cloud_msg.data.front()) + offset;
-  data_ = reinterpret_cast<TT *>(data_char_);
-  data_end_ = reinterpret_cast<TT *>(&(cloud_msg.data.back()) + 1 + offset);
+  std::memcpy(&data_, &data_char_, sizeof data_char_);
+  auto tmpcloud = &(cloud_msg.data.back()) + 1 + offset;
```

Comment:
Also, given how complicated the types are in here, I'd prefer for this to not use `auto` and instead use `TT *` (which I think is the correct type here).

Reply:
You're right it's not very readable (in fact, `auto` here is `U *`). Done in https://github.com/ros2/common_interfaces/pull/79/commits/a2865a8a96ec8aab480ef53082acfc52d224bf1a